### PR TITLE
Move all AMP files into the 'web' directory

### PR DESCRIPTION
**Pull request by OllysCoding**
Part of https://github.com/guardian/dotcom-rendering/issues/7057

Previous PRs:
- Prepare DCRs files: https://github.com/guardian/dotcom-rendering/pull/7795
- Rename 'browser' to 'client': https://github.com/guardian/dotcom-rendering/pull/7799
- Remove 'apps' directory: https://github.com/guardian/dotcom-rendering/pull/7830

This PR removes the 'amp' directory entirely - instead aiming for a 'flat' directory shared between apps, web & amp.

### Why?

Our goal here (with more steps coming in future PRs) is to have a single 'flat' `src` directory in DCR - no longer separated by  AMP, Web or Apps. 

This could enable a future where DCR is built & structure more around the services themselves (e.g articles, fronts, etc), and in the meantime should help project discoverability and enforce more clear practices around where files are placed and how they're named 

**What are the potential advantages of this?**

- _Accidental amp files_: Where files previously had the same name across AR & DCR, using a 'file search' feature, you'd often accidentally end up on the AMP file, missing that they were in the AMP directory - now all amp files end with `.amp.<ext>`, which should make it easier to spot

- _Flatter directory structure_: The flatter structure should aid discoverability, without files hidden in obscure corners

**What are the potential disadvantages of this?**

- _Harder to remove AMP?_: AMP may go away soon, and if it does, previously we could have simply deleted the whole folder - but now the files will be intermingled with others that we want to keep

**What would some other options be?**

- Keep AMP in its own directory, but more all the combined web & apps code into the `src` directory

- Keep AMP in its own directory, and have some kind of 'shared' folder to store both web & apps code-

- Something else? Comment below!

### CODE REVIEW

File: dotcom-rendering/src/types/ampArticleModel.tsx
Comment (JamieB-gu) [inline dotcom-rendering/src/types/ampArticleModel.tsx:1]: `articleModel.amp.ts`?

File: dotcom-rendering/src/web/components/AmpArticlePage.tsx
Comment (JamieB-gu) [inline dotcom-rendering/src/web/components/AmpArticlePage.tsx:1]: `ArticlePage.amp.tsx`?

File: dotcom-rendering/src/web/AMPAttributes.ts
Comment (JamieB-gu) [inline dotcom-rendering/src/web/AMPAttributes.ts:1]: `AMPAttributes.amp.ts`?

File: dotcom-rendering/src/web/components/moustache.amp.tsx
Comment (JamieB-gu) [inline dotcom-rendering/src/web/components/moustache.amp.tsx:1]: Might be a good opportunity to rename to `Moustache`?

File: dotcom-rendering/src/web/server/AMPExperimentCache.ts
Comment (JamieB-gu) [inline dotcom-rendering/src/web/server/AMPExperimentCache.ts:1]: `AMPExperimentCache.amp.ts`?

**Review by JamieB-gu — APPROVED** [review]
LGTM.

Am I correct in thinking that the rule for product (web/apps/amp/editions) suffixes is:

> If this file is meant to be used for a specific product only then add the suffix, else it can be used by multiple products and therefore has no suffix.

?

So a file named `pillar.ts` can be used for multiple products, whereas one called `pillar.web.ts` is for web use only.

I've added a few suggestions to add the `.amp` suffix based on that rule. I know some of them have the word AMP in the main filename, but it's probably good to be consistent so there's no ambiguity?

**Comment by OllysCoding** [comment]
> Perhaps not the place to raise this, but have we ever discussed the impact on AMP on our ad revenue with Commercial? For other reasons in here (https://www.reddit.com/r/AmputatorBot/comments/ehrq3z/why_did_i_build_amputatorbot/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button), AMP remains a controversial option and I think its separation from other parts of our code might be a good thing

Thanks for raising this Parisa, I think it's a really important discussion. I know that outside of this piece of work (and engineering in general) there are other places where this discussion is taking place - worth chatting to Rhianna if you have some insight to add there.

> have we ever discussed the impact on AMP on our ad revenue with Commercial

While this isn't necessarily common for all publishers, as I understand AMP remains _more_ valuable to us than regular pages, likely due to its high ad-ratio and bottom-banner advert.

----

In terms of in relation to this PR, I do agree that separation of our AMP code is important, AMP is essentially its own platform from a technical point. However - having it as a 'outlier' in our code has meant it's generally fallen behind in both understanding & structure, when DCR has evolved, AMP often has not evolved with it. This has left the DCR project with inconsistent practices.

My hope with this PR is that AMP remains adequately separated from our web code (All amp specific files end in `.amp.<ext>`), while still integrated at the file system level in a way which highlights that it should not be entirely ignored or forgotten.

Unfortunately from an engineering perspective the future of AMP remains outside of our control - so for now we should assume we'll have to maintain our AMP codebase for the years to come, and therefor I think getting back on top of our AMP code would be valuable!

**Pull request merged**